base/applications/cmdutils/more/more.c

```diff
@@ -749,44 +749,48 @@ FileGetString(
     return TRUE;
 }
 
-static VOID
-LoadRegistrySettings(HKEY hKeyRoot)
+static BOOL
+LoadRegistrySettings(HKEY hKeyRoot, BOOL bDefault)
 {
     LONG lRet;
     HKEY hKey;
-    DWORD dwType, len;
+    DWORD dwType, dwSize;
+    BOOL ret = bDefault;
     /*
      * Buffer big enough to hold the string L"4294967295",
      * corresponding to the literal 0xFFFFFFFF (MAXULONG) in decimal.
      */
-    DWORD Buffer[6];
+    WCHAR Buffer[11];
+    typedef char assertion_1[(sizeof(Buffer) >= sizeof(L"4294967295")) ? 1 : -1];
+    typedef char assertion_2[(sizeof(Buffer) >= sizeof(DWORD)) ? 1 : -1];
 
     lRet = RegOpenKeyExW(hKeyRoot,
                          L"Software\\Microsoft\\Command Processor",
                          0,
                          KEY_QUERY_VALUE,
                          &hKey);
     if (lRet != ERROR_SUCCESS)
-        return;
+        return ret;
 
-    len = sizeof(Buffer);
+    dwSize = sizeof(Buffer);
     lRet = RegQueryValueExW(hKey,
                             L"EnableExtensions",
                             NULL,
                             &dwType,
                             (LPBYTE)&Buffer,
-                            &len);
+                            &dwSize);
     if (lRet == ERROR_SUCCESS)
     {
         /* Overwrite the default setting */
         if (dwType == REG_DWORD)
-            s_bEnableExtensions = !!*(PDWORD)Buffer;
+            ret = (*(PDWORD)Buffer != 0);
         else if (dwType == REG_SZ)
-            s_bEnableExtensions = (_wtol((PWSTR)Buffer) == 1);
+            ret = (_wtol(Buffer) != 0);
     }
     // else, use the default setting set globally.
 
     RegCloseKey(hKey);
+    return ret;
```

Comment:
Not good, please revert (except for the `DWORD Buffer[6];` --> `WCHAR Buffer[11];` that is correct).

Reply:
Reverted.